### PR TITLE
Prevent 'Null' from being displayed as the users first name

### PR DESCRIPTION
**Pull request by omgitsbillryan**
**GH Issue**
https://github.com/department-of-veterans-affairs/vets.gov-team/issues/2414

### CODE REVIEW

File: test/login/components/SearchHelpSignIn.unit.spec.jsx [hunk 1]
```diff
@@ -4,18 +4,29 @@ import { expect } from 'chai';
 
 import { SearchHelpSignIn } from '../../../src/js/login/components/SearchHelpSignIn.jsx';
 
+const defaultLoginProps = {
```
Comment (omgitsbillryan) [inline test/login/components/SearchHelpSignIn.unit.spec.jsx]: I had to refactor this unit test in order to keep it DRY while adding the one test to cover the `null` greeting fix.

**Review by raquelromano-gov — APPROVED** [review]

File: test/login/components/SearchHelpSignIn.unit.spec.jsx [hunk 2]
```diff
@@ -1,21 +1,33 @@
...
+
+const defaultProfileProps = {
+  email: 'fake@aol.com',
+  userFullName: {
+    first: 'Sharon'
+  }
+};
+
+function buildProps(defaultProps, props = {}) {
+  return _.merge({}, defaultProps, props);
```
Comment (U-DON) [inline test/login/components/SearchHelpSignIn.unit.spec.jsx]: I would suggest using `merge` from `lodash/fp` if you don't want the extra arg to ensure both objects get merged into a new object. The FP version will just create a new object by default.

Comment (omgitsbillryan) [inline test/login/components/SearchHelpSignIn.unit.spec.jsx]: Is there doc that outlines this fact somewhere?  I'm not familiar with the `fp` flavor of lodash.

Comment (U-DON) [inline test/login/components/SearchHelpSignIn.unit.spec.jsx]: It's not a very good doc, but here's the [official FP guide](https://github.com/lodash/lodash/wiki/FP-Guide).

Comment (raquelromano-gov) [inline test/login/components/SearchHelpSignIn.unit.spec.jsx]: @omgitsbillryan @U-DON I believe this doc is auto-generated from the lodash API docs and might be a bit more comprehensive: https://gist.github.com/jfmengels/6b973b69c491375117dc

**Pull request merged**